src/lib/InstantSearch.ts

```diff
@@ -184,6 +186,10 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
       );
     }
 
+    if (!insightsClient && Boolean(getDetectedInsightsClient())) {
```

Comment:
what about:

```js
const possibleDetectedInsightsClient = getDetectedInsightsClient();

if (!insightsClient && possibleDetectedInsightsClient) {
  insightsClient = possibleDetectedInsightsClient;
}
```

since it explicitly returns false, I don't see the need of wrapping it in a boolean call (&& already does it)

Reply:
I should make it return `Function | undefined` instead

Reply:
I changed it to `hasDetectedInsightsClient` so it's even more obvious now that it returns a boolean.